ci-operator/step-registry/assisted/ofcir/setup/assisted-ofcir-setup-commands.sh

```diff
@@ -0,0 +1,144 @@
+  tasks:
+    - name: Check if ansible inventory exists
+      stat:
+        path: "{{ SHARED_DIR }}/inventory"
+      register: inventory
+    - name: Create default ansible inventory
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/inventory"
+        content: |
+          [primary]
+          primary-{{ lookup('env', 'IP') }} ansible_host={{ lookup('env', 'IP') }} ansible_user=root ansible_ssh_user=root ansible_ssh_private_key_file={{ lookup('env', 'SSH_KEY_FILE') }} ansible_ssh_common_args="-o ConnectTimeout=5 -o UserKnownHostsFile=/dev/null -o ServerAliveInterval=90 -o LogLevel=ERROR"
+      when: not inventory.stat.exists
+    - name: Create ssh config file
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/ssh_config"
+        content: |
+          Host ci_machine
+            User root
+            HostName {{ lookup('env', 'IP') }}
+            ConnectTimeout 5
+            StrictHostKeyChecking no
+            ServerAliveInterval 90
+            LogLevel ERROR
+            IdentityFile {{ lookup('env', 'SSH_KEY_FILE') }}
+            ConnectionAttempts 10
+    - name: Create ansible configuration
+      ansible.builtin.copy:
+        dest: "{{ SHARED_DIR }}/ansible.cfg"
+        content: |
+          [defaults]
+          callback_whitelist = profile_tasks
+          host_key_checking = False
+
+          verbosity = 2
+          stdout_callback = yaml
+          bin_ansible_callbacks = True
+
+          [ssh_connection]
+          retries = 10
+EOF
+
+ansible-playbook prepare-ansible.yaml
+export ANSIBLE_CONFIG="${ANSIBLE_CONFIG_FILE}"
+
+cat > setup.yaml <<-"EOF"
+- name: Make sure inventory contains at least one host
+  hosts: localhost
+  tasks:
+    - fail:
+        msg: "[ERROR] Empty inventory. No host available."
+      when: groups.all|length == 0
+      
+- name: Set up remote host
+  hosts: primary
+  vars:
+    CLUSTERTYPE: "{{ lookup('env', 'CLUSTERTYPE') }}"
+    SWAP_FILE_SIZE: "{{ lookup('env', 'SWAP_FILE_SIZE') }}"
+  pre_tasks:
+    - name: wait for ssh
+      ansible.builtin.wait_for_connection:
+        sleep: 30
+        delay: 30
+  tasks:
+    - debug:
+        var: CLUSTERTYPE
+        
+    - name: Setup swap file for small / medium machines
+      ansible.builtin.shell: |
+        sudo fallocate -l {{ SWAP_FILE_SIZE }} /swapfile
+        sudo chmod 600 /swapfile
+        sudo mkswap /swapfile
+        sudo swapon /swapfile
+        echo '/swapfile none swap defaults,pri=10 0 0' | sudo tee -a /etc/fstab
+      when: '"small" in CLUSTERTYPE or "medium" in CLUSTERTYPE'
+
+    - name: Select the largest disk for the working directory
+      ansible.builtin.shell: |
+        ROOT_DISK=$(lsblk -o pkname --noheadings --path | grep -E "^\S+" | sort | uniq)
+        DATA_DISK=$(lsblk -o name --noheadings --sort size --path | grep -v "${ROOT_DISK}" | tail -n1)
+        mkfs.xfs -f "${DATA_DISK}"
+        mount "${DATA_DISK}" {{ REPO_DIR }}
+      when: '"large" in CLUSTERTYPE'
```

Comment:
same, we may want to generalize this setup, we know that test-infra requires about 500GB, so we can select a dynamically a working disk.